src/main/java/space/obminyashka/items_exchange/controller/AdvertisementController.java

```diff
@@ -63,15 +64,29 @@ public Page<AdvertisementTitleDto> findPaginatedAsThumbnails(
             @Parameter(name = "Results page you want to retrieve (0..N). Default value: 0")
             @RequestParam(value = "page", required = false, defaultValue = "0") @PositiveOrZero int page,
             @Parameter(name = "Number of records per page. Default value: 12")
-            @RequestParam(value = "size", required = false, defaultValue = "12") @PositiveOrZero int size) {
+            @RequestParam(value = "size", required = false, defaultValue = "12") @PositiveOrZero int size,
+            @Parameter(name = "ID of existed advertisement")
+            @RequestParam(value = "advertisementId", required = false) UUID advertisementId,
+            @Parameter(name = "ID of existed subcategory")
+            @RequestParam(value = "subcategoryId", required = false) Long subcategoryId) {
+        if (advertisementId != null && subcategoryId != null) {
+            return advertisementService.findAllThumbnails(PageRequest.of(page, size), advertisementId, subcategoryId);
+        }
+
         return advertisementService.findAllThumbnails(PageRequest.of(page, size));
     }
 
     @GetMapping(ApiKey.ADV_THUMBNAIL_RANDOM)
-    @Operation(summary = "Find 12 random advertisement as thumbnails and return them as a result")
+    @Operation(summary = "Find N random advertisement as thumbnails and return them as a result with filters")
     @ApiResponse(responseCode = "200", description = "OK")
-    public List<AdvertisementTitleDto> findRandom12Thumbnails() {
-        return advertisementService.findRandom12Thumbnails();
+    public List<AdvertisementTitleDto> findRandom12Thumbnails(
+            @Parameter(name = "Number of random advertisements. Default value: 12")
+            @RequestParam(value = "amount", required = false, defaultValue = "12") @Positive int amount,
+            @Parameter(name = "ID of existed advertisement")
+            @RequestParam(value = "advertisementId", required = false) UUID advertisementId,
```

Comment:
The variable's name and description don't bring any context to its purpose. What do you think it should do?